src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11573,6 +11572,56 @@ void Assembler::evex_prefix(bool vex_r, bool vex_b, bool vex_x, bool evex_r, boo
   emit_int32(EVEX_4bytes, byte2, byte3, byte4);
 }
 
+void Assembler::ext_evex_prefix(bool vex_r, bool vex_b, bool vex_x, bool evex_r, bool evex_v,
+                       bool eevex_r, bool eevex_b, bool eevex_x,
+                       int nds_enc, VexSimdPrefix pre, VexOpcode opc) {
+  // EVEX 0x62 prefix
+  // byte1 = EVEX_4bytes;
+
+  bool vex_w = _attributes->is_rex_vex_w();
+  int evex_encoding = (vex_w ? VEX_W : 0);
+  // EVEX.b is not currently used for broadcast of single element or data rounding modes
+  _attributes->set_evex_encoding(evex_encoding);
+
+  // P0: byte 2, initialized to RXBR`00mm
+  // instead of not'd
+  int byte2 = (vex_r ? VEX_R : 0) | (vex_x ? VEX_X : 0) | (vex_b ? VEX_B : 0) | (evex_r ? EVEX_Rb : 0) 
+            | (eevex_r ? EEVEX_R : 0) | (eevex_b ? EEVEX_B : 0) | (eevex_x ? EEVEX_X : 0);
+  byte2 = (~byte2) & 0xF0;
+  // confine opc opcode extensions in mm bits to lower two bits
+  // of form {0F, 0F_38, 0F_3A}
+  byte2 |= opc;
+
+  // P1: byte 3 as Wvvvv1pp
+  int byte3 = ((~nds_enc) & 0xf) << 3;
+  // p[10] is always 1
+  // byte3 |= EVEX_F;
+  byte3 |= eevex_x ? 0 : EEVEX_X; // inverted logic
```

Comment:
EEVEX_X is 0x40 which seems to overlap with top bit of vvvv in byte3?

Reply:
Fixed. Thanks.  Should be EEVEX_X = 0x04